arangod/Aql/Functions.cpp

```diff
@@ -6507,3 +6518,412 @@ AqlValue Functions::Fail(arangodb::aql::Query* query, transaction::Methods* trx,
+      }
+      return std::to_string(minutes);
+    }},
+  {"%ss", [](std::string& wrk, tp_sys_clock_ms const& tp) {
+      auto day_time = make_time(tp - floor<days>(tp));
+      uint64_t seconds = day_time.seconds().count();
+      if (seconds < 10) {
+        return std::string("0") + std::to_string(seconds);
+      }
+      return std::to_string(seconds);
+    }},
+  {"%kk", [](std::string& wrk, tp_sys_clock_ms const& tp) {
+      iso_week::year_weeknum_weekday yww{floor<days>(tp)};
+      uint64_t isoWeek = static_cast<uint64_t>((unsigned)(yww.weeknum()));
+      if (isoWeek < 10) {
+        return std::string("0") + std::to_string(isoWeek);
+      }
+      return std::to_string(isoWeek);
+    }},
+  
+  {"%t", [](std::string& wrk, tp_sys_clock_ms const& tp) {
+      auto diffDuration = tp - unix_epoch;
+      auto diff = duration_cast<duration<double, std::milli>>(diffDuration).count();
+      return std::to_string(static_cast<int64_t>(std::round(diff)));
+    }},
+  {"%z", [](std::string& wrk, tp_sys_clock_ms const& tp) {
+      std::string formatted = format("%FT%TZ", floor<milliseconds>(tp));
+      return formatted;
+    }},
+  {"%w", [](std::string& wrk, tp_sys_clock_ms const& tp) {
+      weekday wd{floor<days>(tp)};
+      return std::to_string(static_cast<uint64_t>(unsigned(wd)));
+    }},
+  {"%y", [](std::string& wrk, tp_sys_clock_ms const& tp) {
+      auto ymd = year_month_day(floor<days>(tp));
+      return std::to_string(static_cast<int64_t>((int)year{ymd.year()}));
+    }},
+  {"%m", [](std::string& wrk, tp_sys_clock_ms const& tp) {
+      auto ymd = year_month_day(floor<days>(tp));
+      return std::to_string(static_cast<uint64_t>((unsigned)ymd.month()));
+    }},
+  {"%d", [](std::string& wrk, tp_sys_clock_ms const& tp) {
+      auto ymd = year_month_day(floor<days>(tp));
+      return std::to_string(static_cast<uint64_t>((unsigned)ymd.day()));
+    }},
+  {"%h", [](std::string& wrk, tp_sys_clock_ms const& tp) {
+      auto day_time = make_time(tp - floor<days>(tp));
+      uint64_t hours = day_time.hours().count();
+      return std::to_string(hours);
+    }},
+  {"%i", [](std::string& wrk, tp_sys_clock_ms const& tp) {
+      auto day_time = make_time(tp - floor<days>(tp));
+      uint64_t minutes = day_time.minutes().count();
+      return std::to_string(minutes);
+    }},
+  {"%s", [](std::string& wrk, tp_sys_clock_ms const& tp) {
+      auto day_time = make_time(tp - floor<days>(tp));
+      uint64_t seconds = day_time.seconds().count();
+      return std::to_string(seconds);
+    }},
+  {"%f", [](std::string& wrk, tp_sys_clock_ms const& tp) {
+      auto day_time = make_time(tp - floor<days>(tp));
+      uint64_t millis = day_time.subseconds().count();
+      return std::to_string(millis);
+    }},
+  {"%x", [](std::string& wrk, tp_sys_clock_ms const& tp) {
+      auto ymd = year_month_day(floor<days>(tp));
+      auto yyyy = year{ymd.year()};
+      auto firstDayInYear = yyyy / 1 / 0;
```

Comment:
why intentional division by zero here?

Reply:
https://stackoverflow.com/questions/10672461/how-to-get-the-first-and-the-last-dates-day-and-month-of-a-specified-week-numb explains that this can be used to calculate the day.